src/Resources/views/CRUD/base_array_macro.html.twig

```diff
@@ -8,17 +8,75 @@ For the full copyright and license information, please view the LICENSE
 file that was distributed with this source code.
 
 #}
-{% macro render_array(value, inline) %}
-    {% from _self import render_array %}
-    {% for key, val in value %}
-        {% if val is iterable %}
-            [{{ key }} => {{ render_array(val, inline) }}]
-        {% else %}
-            [{{ key }} => {{ val }}]
-        {% endif %}
-
-        {% if not loop.last and not inline %}
-            <br>
-        {% endif %}
-    {% endfor %}
+{# NEXT_MAJOR: move inline parameter under options #}
+{% macro render_array(value, inline, options = []) %}
+    {%- apply spaceless -%}
+        {%- from _self import render_array -%}
+
+        {%- if not inline -%}
+            <ul>
+        {%- else -%}
+            [
+        {%- endif -%}
+
+        {%- for key, val in value -%}
+            {%- if not inline -%}
+                <li>
+            {%- endif -%}
+
+            {%- if options.display is not defined or options.display in ['both', 'keys'] -%}
+                {%- if options.key_translation_domain is not defined or options.key_translation_domain is same as(false) -%}
```

Comment:
```suggestion
                {%- if options.key_translation_domain|default(false) is same as(false) -%}
```

Reply:
When we pass `'key_translation_domain' => null,` explicitly we will get `options.key_translation_domain is not defined` equals to `false` but `options.key_translation_domain|default(false)` will change our `null` value to `false` and our logic will not work properly.
In case when we don't pass any value to `key_translation_domain`, `options.key_translation_domain is not defined` equals to `true` and everything will work as expected.
 The same for other similar changes.

Reply:
Ok, how about defining default values on line 12? I added a change suggestion.

Reply:
Ok how about defining default values, like in the suggestion above?